File: app/src/main/java/com/mmutert/freshfreezer/ui/databinding/AmountBindingConverter.java
package com.mmutert.freshfreezer.ui.databinding;

import androidx.databinding.InverseMethod;

import java.util.Locale;


public class AmountBindingConverter {

    @InverseMethod("stringToFloat")
    public static String floatToString(float newValue) {
        return String.format(Locale.getDefault(), "%f", newValue);
    }

    public static float stringToFloat(String newValue) {
        try {
            return Float.parseFloat(newValue);
        } catch (NumberFormatException e) {
            return 0.0F;
        }
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/ui/databinding/AmountBindingConverter.kt
package com.mmutert.freshfreezer.ui.databinding

import androidx.databinding.InverseMethod

object AmountBindingConverter {
    @InverseMethod("stringToFloat")
    fun floatToString(newValue: Float): String {
        return newValue.toString()
    }

    fun stringToFloat(newValue: String): Float {
        return try {
            newValue.toFloat()
        } catch (e: NumberFormatException) {
            0.0f
        }
    }
}